src/Setup/BuildPlan.purs

```diff
@@ -2,58 +2,81 @@ module Setup.BuildPlan (constructBuildPlan, BuildPlan) where
 
 import Prelude
 
-import Control.Monad.Except.Trans (ExceptT)
+import Control.Monad.Except.Trans (ExceptT, mapExceptT)
 import Data.Argonaut.Core (Json)
 import Data.Argonaut.Decode (decodeJson, printJsonDecodeError, (.:))
 import Data.Array as Array
 import Data.Bifunctor (lmap)
 import Data.Either (Either(..))
 import Data.Foldable (fold)
 import Data.Maybe (Maybe(..))
+import Data.String (Pattern(..))
+import Data.String as String
 import Data.Traversable (traverse)
 import Data.Version (Version)
 import Data.Version as Version
-import Effect (Effect)
-import Effect.Aff (error, throwError)
+import Effect.Aff (Aff, error, throwError)
+import Effect.Aff.Class (liftAff)
 import Effect.Class (liftEffect)
 import Effect.Exception (Error)
 import GitHub.Actions.Core as Core
 import Setup.Data.Key (Key)
 import Setup.Data.Key as Key
 import Setup.Data.Tool (Tool)
 import Setup.Data.Tool as Tool
+import Setup.UpdateVersions (ReleaseType(..), fetchFromGitHubReleases)
 import Text.Parsing.Parser (parseErrorMessage)
 import Text.Parsing.Parser as ParseError
 
 -- | The list of tools that should be downloaded and cached by the action
 type BuildPlan = Array { tool :: Tool, version :: Version }
 
 -- | Construct the list of tools that sholud be downloaded and cached by the action
-constructBuildPlan :: Json -> ExceptT Error Effect BuildPlan
+constructBuildPlan :: Json -> ExceptT Error Aff BuildPlan
 constructBuildPlan json = map Array.catMaybes $ traverse (resolve json) Tool.allTools
 
 -- | The parsed value of an input field that specifies a version
-data VersionField = Latest | Exact Version
+data VersionField
+  -- | Pre-releases for versions matching the given version
+  = Unstable Version
+  -- | Lookup the latest release that is not a pre-release
+  | Latest
+  -- | Use the given version
+  | Exact Version
 
 -- | Attempt to read the value of an input specifying a tool version
-getVersionField :: Key -> ExceptT Error Effect (Maybe VersionField)
+getVersionField :: Key -> ExceptT Error Aff (Maybe VersionField)
 getVersionField key = do
-  value <- Core.getInput' (Key.toString key)
+  value <- mapExceptT liftEffect $ Core.getInput' (Key.toString key)
   case value of
     "" ->
       pure Nothing
     "latest" ->
       pure (pure Latest)
-    val -> case Version.parseVersion val of
-      Left msg -> do
-        liftEffect $ Core.error $ fold [ "Failed to parse version ", val ]
-        throwError (error (ParseError.parseErrorMessage msg))
-      Right version ->
-        pure (pure (Exact version))
+    val
+      | Just versionStr <- String.stripPrefix (Pattern "unstable-") val -> do
+          if Key.toString key /= "purescript" then do
```

Comment:
I wasn't sure how I should handle this situation since only `purescript` is actually supported. Should `purescriptKey` be exported?